RELEASING.md

```diff
@@ -4,23 +4,31 @@ Releasing checklist:
 
 (1) Write a changelog.
 
-(2) Checkout the branch to be released. This should be `main` except in the event of a hotfix. For hotfixes, checkout the latest release branch.
-
-(3) Create a tag for the release.
+(2) Run version bump script with the new version as an argument and open a PR.
 
 ```sh
-git tag v0.2.0
+python scripts/update-version.py v0.5.1
+```
+
+(3) Create and push a release branch.
+
+```txt
+git checkout -b release-v0.5.1
+git push release-v0.5.1
```

Comment:
What is the purpose of pushing a release branch? Is it not enough with pushing the tag from main?

Reply:
Great question! I think the main reason lies in having an easy reference to use/checkout in the event of a hotfix. I know that I pull up release branches when looking up version-specific questions as a matter of convenience as well

Reply:
Couldn't we use/checkout the tag reference for the usage you are describing? In the end, a branch is just a tag to a commit with the difference that it "advances" and in this case we don't need the release branch to advance further. I still feel is enough pushing the tag for that.

Reply:
might be enough. the other reason was how the [docsite versioning](https://github.com/OpenZeppelin/docs.openzeppelin.com/blob/master/playbook.yml#L70) works (it uses branches to look for previous versions docs) but it can be [changed to tags](https://docs.antora.org/antora/latest/playbook/configure-content-sources/#content-key).